manifests.go

```diff
@@ -0,0 +1,66 @@
+package distribution
+
+import (
+	"github.com/docker/distribution/context"
+	"github.com/docker/distribution/digest"
+	"github.com/docker/distribution/reference"
+)
+
+// Manifest represents a registry object specifying a target and a set of constituents
+type Manifest interface {
+	// Target returns a descriptor for the configuration of this manifest.  This
+	// may return a nil descriptor if none exists for this manifest.
+	Target() Descriptor
+
+	// Constituents returns a list of objects which make up this manifest.
+	// The dependencies are strictly ordered from base to head. A constituent
+	// is anything which can be represented by a distribution.Descriptor
+	Constituents() []Descriptor
```

Comment:
Sweet! :bike: :house_with_garden: initiated! `<bikeshedding music plays, mostly death metal cause its going to be brutal!>` :guitar: :bike: :house_with_garden: :guitar:

I like `References`. I think that captures the intent.